ui/org.eclipse.pde.launching/src/org/eclipse/pde/internal/launching/JUnitLaunchRequirements.java

```diff
@@ -0,0 +1,164 @@
+			plugins.add(JUNIT4_RUNTIME_PLUGIN);
+		} else if (org.eclipse.jdt.internal.junit.launcher.TestKindRegistry.JUNIT5_TEST_KIND_ID.equals(testKind.getId())) {
+			plugins.add(JUNIT5_RUNTIME_PLUGIN);
+		}
+		return plugins;
+	}
+
+	private static void addPlugins(Collection<String> plugins, Map<String, List<IPluginModelBase>> allBundles, Map<IPluginModelBase, String> allModels) throws CoreException {
+		Set<String> requiredPlugins = new LinkedHashSet<>(plugins);
+
+		Set<BundleDescription> addedRequirements = new LinkedHashSet<>();
+		addAbsentRequirements(requiredPlugins, addedRequirements, allBundles, allModels);
+
+		Set<BundleDescription> requirementsOfRequirements = DependencyManager.findRequirementsClosure(addedRequirements);
+		addAbsentRequirements(requirementsOfRequirements, allBundles, allModels);
+	}
+
+	private static void addAbsentRequirements(Collection<String> requirements, Set<BundleDescription> addedRequirements, Map<String, List<IPluginModelBase>> allBundles, Map<IPluginModelBase, String> allModels) throws CoreException {
+		for (String id : requirements) {
+			List<IPluginModelBase> models = allBundles.computeIfAbsent(id, k -> new ArrayList<>());
+			if (models.stream().noneMatch(m -> m.getBundleDescription().isResolved())) {
+				IPluginModelBase model = JUnitLaunchRequirements.findRequiredPluginInTargetOrHost(id);
+				models.add(model);
+				BundleLauncherHelper.addDefaultStartingBundle(allModels, model);
+				if (addedRequirements != null) {
+					addedRequirements.add(model.getBundleDescription());
+				}
+			}
+		}
+	}
+
+	private static void addAbsentRequirements(Set<BundleDescription> toAdd, Map<String, List<IPluginModelBase>> allBundles, Map<IPluginModelBase, String> allModels) throws CoreException {
+		for (BundleDescription requirement : toAdd) {
+			String id = requirement.getSymbolicName();
+			List<IPluginModelBase> models = allBundles.computeIfAbsent(id, k -> new ArrayList<>());
+			boolean replace = !models.isEmpty() && models.stream().noneMatch(m -> m.getBundleDescription().getVersion().equals(requirement.getVersion()));
+			if (replace || models.stream().noneMatch(m -> m.getBundleDescription().isResolved())) {
+				IPluginModelBase model = JUnitLaunchRequirements.findRequiredPluginInTargetOrHost(requirement);
+				if (replace) {
+					String startLevel = null;
+					for (IPluginModelBase m : models) {
+						startLevel = allModels.remove(m);
+					}
+					models.clear();
+					allModels.put(model, startLevel);
+				}
+				models.add(model);
+				BundleLauncherHelper.addDefaultStartingBundle(allModels, model);
+			}
+		}
+	}
+
+	public static Set<BundleDescription> junit5PlatformRequirements() throws CoreException {
+		// add launcher and jupiter.engine to support @RunWith(JUnitPlatform.class)
+		String[] requiredPlugins = {"junit-platform-launcher", //$NON-NLS-1$
+				"junit-jupiter-engine", //$NON-NLS-1$
+		};
```

Comment:
This could be added as a constant at the top and you could just make this a `List<String>` using `List.of(...)`.